[Shi Kexin] iP

cashewnade opened the pull request:
### Aeromon

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))


Aeromon frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
-  ~~FAST~~ _SUPER_ FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2122s2.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [x]  Managing tasks
- [ ]  Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main `method:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

cashewnade committed:
Level-1

Completed Level-1

cashewnade committed:
Level-2

Completed Level-2. Changed the greeting and farewell message, as well as renamed the chatbot.

cashewnade committed:
Level-3

Completed Level-3

cashewnade committed:
Level-4

Completed Level-4

cashewnade committed:
A-TextUiTesting

Completed TextUiTesting

cashewnade committed:
Level-5

Completed Level-5

cashewnade committed:
Level-6

Completed Level-6

cashewnade committed:
Javadocs

Updated javadocs

cashewnade committed:
Cleanup

Cleaned up code by using switch and case statements

cashewnade committed:
Add feature to save tasks locally

cashewnade committed:
Add features to read dates for Deadline and Todo

cashewnade committed:
Merge branch 'branch-Level-7'

cashewnade committed:
Merge branch 'branch-Level-8'

cashewnade committed:
Edit features to read dates

cashewnade committed:
Refactor the code to extract out closely related code as classes.

cashewnade committed:
Merge branch 'HEAD'

cashewnade committed:
Organize the classes into suitable java packages.

cashewnade committed:
Merge branch 'HEAD'

cashewnade committed:
Fix bugs when committing and pushing

janald99 commented on src/main/java/aeromon/Storage.java:
```diff
@@ -0,0 +1,76 @@
+                while (fileScanner.hasNextLine()) {
+                    String content = fileScanner.nextLine();
+                    String[] info = content.split(" / ");
+                    switch (info[0]) {
```
Perhaps a short comment can be placed here to describe what info[0] refers to?

janald99 commented on src/main/java/aeromon/Storage.java:
```diff
@@ -0,0 +1,76 @@
+                            break;
+                        }
+                        case "D": {
+                            Deadline task = new Deadline(info[2], LocalDate.parse(info[3], dateTimeFormatter));
```
Perhaps the second line could be indented further for better readability?
I noticed the same issue in several other places too.

janald99 commented on src/main/java/aeromon/command/Command.java:
```diff
@@ -0,0 +1,14 @@
+    public abstract void execute(TaskArrayList taskArrayList, Ui ui, Storage storage) throws AeromonException;
+
+    public boolean isExit() {
+        return this instanceof ByeCommand;
```
I like the way you defined the isExit() method. It is often that programmers simply return true or false, but this way of defining it provides readers with an idea how of you are going to use this method as an exit point for the chatbot. 😄

janald99 commented on src/main/java/aeromon/command/CommandManager.java:
```diff
@@ -0,0 +1,87 @@
+        String[] arr = fullCommand.split(" ", 2);
+        String command = arr[0];
+
+        switch (command) {
```
The switch case indentations do not comply with the coding standards of this module. Maybe when you are free you could edit this 😸

janald99 commented on src/main/java/aeromon/command/EditCommand.java:
```diff
@@ -0,0 +1,52 @@
+
+import aeromon.AeromonException;
+import aeromon.Storage;
+import aeromon.Ui;
```
Maybe the aeromo.Ui import can swap with the aermon.task imports? So that the package imports will be in alphabetical order 😉

janald99 commented on src/main/java/aeromon/command/ListCommand.java:
```diff
@@ -0,0 +1,13 @@
+import aeromon.Ui;
+import aeromon.task.TaskArrayList;
+
+public class ListCommand extends Command {
```
Maybe a blank line here could be inserted for better readability?

janald99 reviewed (commented):
LGTM. Just some nits to fix like switch-case indentations. Great job! 😉

WJunHong commented on src/main/java/aeromon/command/AddCommand.java:
```diff
@@ -0,0 +1,52 @@
+
+public class AddCommand extends Command {
+    private TaskType taskType;
+    private String[] info;
```
Maybe can change the name of the field to tokens because info is not really descriptive enough for an array name

WJunHong commented on src/main/java/aeromon/command/ByeCommand.java:
```diff
@@ -0,0 +1,13 @@
+public class ByeCommand extends Command {
+    @Override
+    public void execute(TaskArrayList taskArrayList, Ui ui, Storage storage) throws AeromonException {
+        ui.print("Buai Buai! Ciao for now!");
```
haha that's cute

WJunHong reviewed (commented):
Overall, I like the progress you have made on the project, can't wait to see the full product!

cashewnade committed:
Add JUnit tests to test the behavior of the code

cashewnade committed:
Add JavaDoc comments to the code

cashewnade committed:
Tweak the code to comply with a coding standard.

cashewnade committed:
Give users a way to find a task by searching for a keyword.

cashewnade committed:
Add Gradle support

cashewnade committed:
Merge branch 'add-gradle-support'

cashewnade committed:
Use checkStyle to detect coding style violations.

cashewnade committed:
Merge branch 'branch-A-checkstyle'

cashewnade committed:
Merge remote-tracking branch 'origin/add-gradle-support' into add-gradle-support

cashewnade committed:
Add GUI to Aeromon

cashewnade committed:
Merge branch 'branch-Level-10'

cashewnade committed:
Add Assertions

Use assert feature (not JUnit assertions) to document important assumptions that should hold at various points in the code.

cashewnade committed:
Merge branch 'branch-A-Assertions'

cashewnade committed:
Improve Code Quality

Improve code quality

cashewnade committed:
Merge branch 'branch-A-CodeQuality'

cashewnade committed:
Delete Ui Class

GUI now handles the UI of the Aeromon bot, which makes the original text-based UI redundant.
Let's delete the redundant Ui class and remove all the Ui instances in the Aeromon class.
Removing the class will improve the code quality as we delete unused code.

cashewnade committed:
Detect Duplicates

Add the ability to recognize and deal with duplicate items. e.g., the same task added multiple times.

cashewnade committed:
Revert "Detect Duplicates"

This reverts commit 43193c3f7f0193656e79b0564ad096e3f38b0e71.

cashewnade committed:
Detect Duplicates

Add the ability to recognize and deal with duplicate items. e.g., the same task added multiple times.

cashewnade committed:
Merge pull request #2 from cashewnade/branch-C-DetectDuplicates

Branch c detect duplicates

cashewnade committed:
Merge remote-tracking branch 'origin/master'

cashewnade committed:
Finalize Features

cashewnade committed:
Updated code based on Code Quality Feedback

cashewnade committed:
Add a User Guide

cashewnade committed:
Set theme jekyll-theme-leap-day

cashewnade committed:
Updated User Guide

cashewnade committed:
Merge remote-tracking branch 'origin/master'

cashewnade committed:
Fix minor bugs and add feature to exit GUI automatically

cashewnade committed:
Improve the GUI to make it more polished

cashewnade committed:
Update User Guide to fix typo and add more details
